### PR TITLE
plugin service

### LINKED ISSUE #118: Describe Connector Plugin API
### Feature description

Need an API endpoint that will allow the developer to know what connectors and capabilities of the connector exist.

- [x] proto design https://github.com/ConduitIO/conduit/pull/268
- [x] orchestration
- [x] implementation https://github.com/ConduitIO/conduit/pull/311

### PR DESCRIPTION
**Pull request by maha-hajja**
### Description

plugin service
plugin list API {GET /v1/plugins} or {GET /v1/plugins?name=s.*}
plugin orchestrator
mocks
mini proto changes
added "type" to plugin.parameter

Fixes #118 

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [x] I have written unit tests.
- [ ] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

**Review by lovromazgon — CHANGES_REQUESTED** [review]
Posted lots of comments but nothing major, generally it looks very good 👏  I would love to see some tests for the handler!

File: pkg/plugin/acceptance_testing.go [hunk 1]
```diff
@@ -97,12 +97,12 @@ func testSpecifier_Specify_Success(t *testing.T, tdf testDispenserFunc) {
 		Version:     "v1.2.3",
 		Author:      "Donald Duck",
 		SourceParams: map[string]Parameter{
-			"param1.1": {Default: "foo", Required: false, Description: "Param 1.1 description"},
-			"param2.1": {Default: "bar", Required: true, Description: "Param 1.2 description"},
+			"param1.1": {Default: "foo", Description: "Param 1.1 description"},
+			"param2.1": {Default: "bar", Description: "Param 1.2 description", Validations: []Validation{{VType: ValidationTypeRequired}}},
```
Comment (lovromazgon) [inline pkg/plugin/acceptance_testing.go:101]: We need to set the `Type` in the expectation, then tests should pass (well, and we need to address [this comment](https://github.com/ConduitIO/conduit/pull/311#discussion_r832454638), to make sure `Validations` are actually `nil` if there are none).
```suggestion
			"param1.1": {Default: "foo", Type: "string", Description: "Param 1.1 description"},
			"param2.1": {Default: "bar", Type: "string", Description: "Param 1.2 description", Validations: []Validation{{VType: ValidationTypeRequired}}},
```

File: pkg/web/api/toproto/plugin.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	for _, v := range in {
+		out = append(out, &apiv1.PluginSpecifications_Parameter_Validation{
+			Type:  PluginValidationType(v.VType),
+			Value: v.Value,
+		})
+	}
+	return out
+}
+
+func PluginValidationType(in plugin.ValidationType) apiv1.PluginSpecifications_Parameter_Validation_Type {
```
[the remark this answers is not in the document]
Comment (maha-hajja) [reply pkg/web/api/toproto/plugin.go:58]: @lovromazgon will do! I noticed that connectors and processors use this trick.. but why not the pipelines?

[the remark this answers is not in the document]
Comment (lovromazgon) [reply pkg/web/api/toproto/plugin.go:58]: There the enums don't match 1 to 1, we have 2 different internal states for stopped, while the API only exposes a single stopped state.

**Review by lovromazgon — CHANGES_REQUESTED** [review]
The API plugin error function and handler test need some minor changes, after that I think we're good!

File: pkg/web/api/plugin_v1_test.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	psMock := mock.NewPluginOrchestrator(ctrl)
+	api := NewPluginAPIv1(psMock)
+	logger := log.New(zerolog.Nop())
+
+	pluginService := plugin.NewService(
+		builtin.NewRegistry(logger, builtin.DefaultDispenserFactories...),
+		standalone.NewRegistry(logger),
+	)
+
+	var plist []*apiv1.PluginSpecifications
```
[the remark this answers is not in the document]
Comment (maha-hajja) [reply pkg/web/api/plugin_v1_test.go]: it was deleted on the next commit

[the remark this answers is not in the document]
Comment (lovromazgon) [reply pkg/web/api/plugin_v1_test.go]: Oh, my bad, I must have been looking at the changes without the last commit!

**Review by lovromazgon — APPROVED** [review]
Looks great 👍  Let's wait for Github actions to start running again and see if the build succeeds, then we can go ahead and merge this!

**Pull request merged**